crates/transaction-pool/src/pool/parked.rs

```diff
@@ -79,10 +87,98 @@ impl<T: ParkedOrd> ParkedPool<T> {
         Some(tx.transaction.into())
     }
 
-    /// Removes the worst transaction from this pool.
-    pub(crate) fn pop_worst(&mut self) -> Option<Arc<ValidPoolTransaction<T::Transaction>>> {
-        let worst = self.best.iter().next().map(|tx| *tx.transaction.id())?;
-        self.remove_transaction(&worst)
+    /// Get transactions by sender
+    pub(crate) fn get_txs_by_sender(
+        &self,
+        sender: &Address,
+    ) -> Vec<Arc<ValidPoolTransaction<T::Transaction>>> {
+        self.by_id
+            .values()
+            .filter(|tx| tx.transaction.sender() == *sender)
+            .map(|tx| tx.transaction.clone().into())
+            .collect()
```

Comment:
this can be improved by only checking the sender id, like we do here:

https://github.com/paradigmxyz/reth/blob/main/crates/transaction-pool/src/pool/txpool.rs#L1153-L1160

Reply:
good idea, updated